client/CPlayerInterface.cpp

```diff
@@ -658,7 +659,7 @@ void CPlayerInterface::battleStart(const BattleID & battleID, const CCreatureSet
 {
 	EVENT_HANDLER_CALLED_BY_CLIENT;
 
-	bool useQuickCombat = settings["adventure"]["quickCombat"].Bool();
+	bool useQuickCombat = settings["adventure"]["quickCombat"].Bool() || GAME->map().getMap()->battleOnly;
```

Comment:
Shouldn't it be the opposite? 

If game is in battle-only mode, never use quick combat and always manual combat?

Reply:
I prefer it this way because you can check if you are better (with fewer losses) than AI.

Reply:
Yes. For first time. But I think it's easy to understand the second time.

Fheroes2 uses the same approach.

Reply:
May add a checkbox to disable/enable this in a later PR.